packages/odyssey-react/src/components/Status/Status.tsx

```diff
@@ -0,0 +1,74 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import type { ComponentProps, ComponentPropsWithoutRef } from "react";
+import { forwardRef } from "react";
+import { useCx, withStyles } from "../../utils";
+import styles from "./Status.module.scss";
+
+import ScreenReaderText from "../ScreenReaderText";
+
+export type StatusVariants = "neutral" | "success" | "caution" | "danger";
+
+interface Props
+  extends Omit<ComponentPropsWithoutRef<"div">, "style" | "className"> {
+  /**
+   * The status label.
+   */
+  label: string;
+
+  /**
+   * Visually hides the status label.
+   */
+  labelHidden?: boolean;
+
+  /**
+   * The description of the present state/status.
+   */
+  descriptor: string;
+
+  /**
+   * The visual variant to be displayed to the user.
+   * @default neutral
+   */
+  variant?: StatusVariants;
+}
+
+/**
+ * Status is used to inform users by providing feedback on system states. Status can display broad
+ * operational states as well as granular states like user status.
+ *
+ * @component
+ * @example <Status label={label} descriptor={descriptor} />
+ */
+let Status = forwardRef<HTMLDivElement, Props>((props, ref) => {
+  const { label, descriptor, labelHidden = false, variant = "neutral" } = props;
+
+  const valueClass = useCx(styles.value, styles[`${variant}Variant`]);
+
+  return (
+    <div className={styles.status} role="status" ref={ref}>
+      {!labelHidden && <span className={styles.label}>{label}</span>}
+      {labelHidden && <ScreenReaderText>{label}</ScreenReaderText>}
```

Comment:
this type of logic can be removed entirely if we allow the `label` prop to accept `ReactNode` so that users can handle applying their own SR labels. Even without that suggested change, the logic here is clunky and we can consider adding a separate variable with ternary expression to move this logic outside the returned JSX expression:

```
const label = labelHidden ? <span className={styles.label} children={label} /> : <ScreenReaderText children={label} />

// and then within the JSX

{label}
```

Reply:
`<span className={styles.label}>{label}</span>`

This `span` has visible styles applied. Even if the child content is hidden, the `span` is not `:empty`, so it appears on the page. This is why `labelHidden` removes the parent container and replaces it with `<ScreenReaderText>` rather than wrapping `{ label }` within the same parent.

Reply:
I'm not arguing against the allowance of composition with the label, I'm pointing out that the `span` itself needs to be removed when a visible label is absent or it will have visual repercussions when rendered. 

VRT would have caught this, but I'm raising it early.

Reply:
100% agreed there @jeffbelser-okta!